edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/CertificateRenewal.cs

```diff
@@ -55,8 +57,28 @@ public void Dispose()
 
         void Callback(object _state)
         {
-            this.logger.LogInformation("Restarting process to perform server certificate renewal.");
-            this.cts.Cancel();
+            TimeSpan timeToExpire = certificates.ServerCertificate.NotAfter - DateTime.UtcNow;
```

Comment:
I realize there is duplicated code here, but I couldn't find a clean way to break it apart. The various log lines require some of the calculated values. I'm open to suggestions, but I think the duplication may be the cleanest option.

Reply:
Maybe we can have a method that returns timer duration in TimeSpan, and schedule the timer if the value is > TimeSpan.Zero